Makefile

```diff
@@ -23,7 +23,7 @@ PROMETHEUS_OPERATOR_TAG ?= v0.39.0
 PROMETHEUS_RULES_CRD ?= https://raw.githubusercontent.com/coreos/prometheus-operator/${PROMETHEUS_OPERATOR_TAG}/example/prometheus-operator-crd/monitoring.coreos.com_prometheusrules.yaml
 PROMETHEUS_SERVICE_MONITORS_CRD ?= https://raw.githubusercontent.com/coreos/prometheus-operator/${PROMETHEUS_OPERATOR_TAG}/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml
 ES_OPERATOR_IMAGE ?= quay.io/openshift/origin-elasticsearch-operator:4.4
-SDK_VERSION=v0.15.1
+SDK_VERSION=v0.18.2
```

Comment:
`require.NotNil(t, "", "Is '' nil?")` does *not* fail, ref https://github.com/jaegertracing/jaeger-operator/pull/1126#pullrequestreview-450155969

Should I also remove them or replace with `NotEmpty`?

Reply:
I don't remember how the implementation looks like, but if it can be empty, then add a `NotEmpty`. It it's always a valid namespace name (based on UUID), then you can just remove it.

Reply:
It cannot fail in the current implementation
```
	// Context is used among others for namespace names where '/' is forbidden and must be 63 characters or less
	id := "osdk-e2e-" + uuid.New()
```

*but* `GetID` is not documented, and it does not ensure the namespace actually exists. 

`GetWatchNamespace`/`GetOperatorNamespace` *seems* like better alternatives, but they are not idempotent and returns an error if it already exists, so they are a bit awkward to use from multiple call sites. I'm not what triggers the namespace creation currently :(

```
// TODO: remove before 1.0.0
// Deprecated: GetNamespace() exists for historical compatibility.
// Use GetOperatorNamespace() or GetWatchNamespace() instead
func (ctx *Context) GetNamespace() (string, error)
```

Reply:
Maybe that can be fixed in another PR, I do not have much extra time today.

Reply:
> but they are not idempotent and returns an error if it already exists

that's exactly the reason I moved to `GetID`.